src/app/domain/geolocation/tracking/tracking.js

```diff
@@ -252,10 +256,16 @@ export const createNewStartPoint = (previousPoint, nextPoint) => {
     nextPoint.coords.speed > 0 ? nextPoint.coords.speed : AVG_WALKING_SPEED
   // Calculate the time to subtract based on speed and distance
   const timeToSubtract = (distance / speed) * 1000
-  const newTime = date.getTime() - timeToSubtract
-  if (newTime > new Date(previousPoint.timestamp).getTime()) {
+  let newTime = date.getTime() - timeToSubtract
+
+  // Check the new time is greater than the previous location point, plus 1s for safety.
+  // Here, we want to prevent overlaping points, that might results in overlapping transitions.
+  if (newTime > new Date(previousPoint.timestamp).getTime() + 1000) {
     // Subtract the time from the date
-    date.setTime(date.getTime() - timeToSubtract)
+    date.setTime(newTime)
+  } else {
+    newTime = date.getTime() - 1000
```

Comment:
I don't get this one. Shouldn't it be `+1000` if we want to be sure `newTime` is superior to previous point?

Reply:
No, let me explain a bit more: let's say you have P1 (the previous point, i.e. the last point of the previous trip) and P3 (the first point of the new trip). Here we try to compute P2, as the tracking is often a bit late to wake-up (especially on iOS) and P3 might not represent to actual start of the trip.
So, temporally speaking P1 < P2 < P3.

Here, `date` is actually the one from P3. We try to compute the a `newTime`, relying the on P3 speed. We check if `newTime` is greater than P1 + 1s. If it is not the case, we give up relying on the speed, and set:  P2  = P3 - 1s.  

That being said, we might have a problem if P1 +1s > P3. But if those points are temporally very close, P2 is not useful anyway (and will be filtered by the server).